### PR TITLE
Add secrets access to task roles.

### PR DESCRIPTION
**Pull request by vincentwschau**
Note on the circular dependency that required the use of a new local variable for service names:

The secret roles I added need to loop over each service, and are used to create the ECS task role. The ECS task role is passed into the environment variables for `roundtable` in the service. If the secret roles depended on the `services` local, this causes a circular dependency.

This is good for the future as well, if we need to create any other resource per service, we should only loop over the names of the services if we don't need any of the details inside `local.services`, as there are more and more dependencies / data in `local.services`.

### CODE REVIEW

**Review by dydxwill — APPROVED** [review]

File: indexer/locals.tf [hunk 1]
```diff
@@ -5,9 +5,24 @@ locals {
   rds_port      = 5432
 }
 
+locals {
+  // Needed so that there are no circular dependencies for all resources are created per service names.
+  service_names = {
```
Comment (Christopher-Li) [inline indexer/locals.tf:10]: Can't we just use a list here?

File: indexer/locals.tf [hunk 2]
```diff
@@ -148,15 +163,10 @@ locals {
...
-          {
-            name : "PG_POOL_MAX",
-            value : "2"
-          },
-          var.roundtable_ecs_environment_variables,
         ],
       ),
     },
-    "vulcan" : {
+    "${local.service_names["vulcan"]}" : {
```
Comment (Christopher-Li) [inline indexer/locals.tf:169]: What's the purpose of this?

File: indexer/locals.tf [hunk 3]
```diff
@@ -148,15 +163,10 @@ locals {
             name : "BLOCKED_ADDRESSES",
             value : var.indexer_compliance_blocklist,
           },
-          {
-            name : "PG_POOL_MAX",
-            value : "2"
-          },
-          var.roundtable_ecs_environment_variables,
```
Comment (Christopher-Li) [inline indexer/locals.tf:155]: Why remove this?

File: indexer/locals.tf (hunk 1 above)
Comment (vincentwschau) [reply indexer/locals.tf:10]: No, it'll make it unclear when we're using it in the `services` local variable below (e.g. service_names[0] is much less clear than service_names["ender"]).

File: indexer/locals.tf (hunk 2 above)
Comment (vincentwschau) [reply indexer/locals.tf:169]: To make it clear that the service names are from the locals variable `service_names`.

File: indexer/locals.tf (hunk 3 above)
Comment (vincentwschau) [reply indexer/locals.tf]: Incorrect resolved a merge conflict. fixed.

**Review by Christopher-Li — APPROVED** [review]

**Pull request merged**